src/js/iqons.js

```diff
@@ -137,6 +163,133 @@ export default class Iqons {
         );
     }
 
+    // Generate a nice tuple of (adjective, verb, noun) describing the Nimiqon.
+    //  - The *Nr are indices of word families/modifiers (see catalog.html)
+    //  - entropy1 decides which feature (face, top, side, bottom) is included as which word type
+    //  - entropy2 decides which variation of the max 27 legal tuples is chosen
+    // Returns: [oneLineString, twoLineString] by calling _wordRound until it succeeds
+    static _words(faceNr, topNr, sidesNr, bottomNr, entropy1, entropy2) {
+        entropy1 = Number(entropy1);
+        entropy2 = Number(entropy2);
+
+        for (let i = 0; i < 100; i++) {
+            const _e1 = entropy1 + i;
+            const result = Iqons._wordRound(faceNr, topNr, sidesNr, bottomNr, _e1, entropy2);
+            if (result) return result;
+        }
+
+        return ["Rare Shiny Bug", "Rare\nShiny Bug"];
+    }
+
+    // Returns: [oneLineString, twoLineString] or null
+    static _wordRound(faceNr, topNr, sidesNr, bottomNr, entropy1, entropy2) {
+        // The catalog contains four features: Faces, Tops, Sides and Bottoms.
+        // Each feature has a list of 21 word families.
+        // We select the same word families as shown in the identicon.
+        const getFamily = (catID, i) => {
+            const cat = IqonsCatalog[catID];
+            return cat[i % cat.length];
+        }
+
+        // Each word family consists of nine entries,
+        //   the 1st three are adjectives,
+        //   the 2nd three are verbs,
+        //   the 3rd three are nouns
+        const _adjectives = a => a.slice(0, 3);
+        const _verbs      = a => a.slice(3, 6);
+        const _nouns      = a => a.slice(6, 9);
+
+        // Build array of all possible word lists
+        //   (string[9])[4]
+        const wordLists = [
+            getFamily("face", faceNr),
+            getFamily("top", topNr),
+            getFamily("side", sidesNr),
+            getFamily("bottom", bottomNr)
+        ]; 
```

Comment:
`_wordRound` should almost never be called more than once, the for loop in `_words` looks a bit misleading.
I removed `getFamily` anyways because there's no point, this is way cleaner: 👍
```js
const wordLists = [
    IqonsCatalog["face"  ][faceNr   % 21],
    IqonsCatalog["top"   ][topNr    % 21],
    IqonsCatalog["side"  ][sidesNr  % 21],
    IqonsCatalog["bottom"][bottomNr % 21]
];
```